### PR TITLE
[WIP] Use hydra structured config

### PR DESCRIPTION
**Pull request by Kshitij09**
This PR involves experimental code for [hydra](https://github.com/facebookresearch/hydra)'s next release

Particulary, I've explored the ways to make use of `TargetConf` and `ConfigStore` for defining [albumentation](https://github.com/albumentations-team/albumentations/)'s transforms

The [peddet/transforms.py](peddet/transforms.py) has the original transforms definition and has been provided for the reference.

The main code for the review has been defined inside [conf/augmentations.py](conf/augmentations.py)

When I run the [test_main.py](test_main.py), the hydra is throwing following error:

```python
Traceback (most recent call last):
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/utils.py", line 203, in run_and_report
    return func()
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/utils.py", line 352, in <lambda>
    lambda: hydra.run(
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/hydra.py", line 98, in run
    cfg = self.compose_config(
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/hydra.py", line 507, in compose_config
    cfg = self.config_loader.load_configuration(
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/config_loader_impl.py", line 148, in load_configuration
    return self._load_configuration(
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/config_loader_impl.py", line 250, in _load_configuration
    cfg = self._merge_defaults_into_config(
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/config_loader_impl.py", line 797, in _merge_defaults_into_config
    hydra_cfg = merge_defaults_list_into_config(hydra_cfg, user_list)
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/config_loader_impl.py", line 769, in merge_defaults_list_into_config
    merged_cfg = self._merge_config(
  File "/home/kshitij/anaconda3/envs/pytorch/lib/python3.8/site-packages/hydra/_internal/config_loader_impl.py", line 707, in _merge_config
    raise MissingConfigException(msg, new_cfg, options)
hydra.errors.MissingConfigException: Could not load aug/to_tensor
```

### CODE REVIEW

File: conf/augmentations.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+from hydra.core.config_store import ConfigStore
```
Comment (omry) [inline conf/augmentations.py:1]: I think your problem is that you are not importing this module so the configs never gets registered.

Comment (Kshitij09) [reply conf/augmentations.py:1]: Fixed it!

**Comment by Kshitij09** [comment]
@omry how do I print the entire config group and not just the default ones?

**Comment by omry** [comment]
> @omry how do I print the entire config group and not just the default ones?

You don't.
you can compose a config an print it with `--cfg job`

**Pull request closed without merging**